bedrock/releasenotes/tests/test_models.py

```diff
@@ -119,6 +120,18 @@ def test_field_processors(self):
         assert note.note.startswith("<p>Firefox Nightly")
         assert note.id == 787203
 
+    def test_product_method_gets_specifically_latest_esr_based_on_product_details(self):
+        _patched_dict = product_details.firefox_versions
+        _patched_dict.update({"FIREFOX_ESR": "999.76"})
+
+        with patch.dict("bedrock.releasenotes.models.product_details.firefox_versions", _patched_dict):
+            # See https://github.com/mozilla/bedrock/issues/16289
+            query = models.ProductRelease.objects.product(product_name="firefox", channel_name="esr")
+
+        raw_query_as_str = str(query.query)
+        assert 'AND "releasenotes_productrelease"."channel" LIKE esr' in raw_query_as_str
+        assert 'AND "releasenotes_productrelease"."version" = 999.7' in raw_query_as_str
```

Comment:
Is the truncated version here (999.7 vs 999.76 set above) meant to demonstrate something? (If so, the idea could be helpful as a comment, to show why these two actually differ, if that's intentional.)

Reply:
Fixed!